[BUG] DLQ Serialization Doesn't Support Create Action

**Describe the bug**
DataPrepper supports `create` and `index` actions in the OpenSearch sink. Depending on which action is specified, the BulkRequest populates the respective parameter. This is accounted for in some places of the code, such as the [JavaClientUncompressedAccumulatingBulkRequest](https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/bulk/JavaClientAccumulatingUncompressedBulkRequest.java#L75-L78), but is not accounted for in the DLQ serialization logic:
https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/dlq/FailedBulkOperationConverter.java#L65

This causes an exception that shuts down DataPrepper:
```
ERROR org.opensearch.dataprepper.pipeline.common.FutureHelper - FutureTask failed due to: 
java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Cannot get 'Index' variant: current variant is 'Create'.
    at java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[?:?]
    at java.util.concurrent.FutureTask.get(FutureTask.java:191) ~[?:?]
    at org.opensearch.dataprepper.pipeline.common.FutureHelper.awaitFuturesIndefinitely(FutureHelper.java:29) ~[data-prepper-core-2.3.1.jar:?]
    at org.opensearch.dataprepper.pipeline.ProcessWorker.postToSink(ProcessWorker.java:140) ~[data-prepper-core-2.3.1.jar:?]
    at org.opensearch.dataprepper.pipeline.ProcessWorker.doRun(ProcessWorker.java:121) ~[data-prepper-core-2.3.1.jar:?]
    at org.opensearch.dataprepper.pipeline.ProcessWorker.run(ProcessWorker.java:50) ~[data-prepper-core-2.3.1.jar:?]
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
    at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
    at java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: java.lang.IllegalStateException: Cannot get 'Index' variant: current variant is 'Create'.
    at org.opensearch.client.util.TaggedUnionUtils.get(TaggedUnionUtils.java:47) ~[opensearch-java-2.5.0.jar:?]
    at org.opensearch.client.opensearch.core.bulk.BulkOperation.index(BulkOperation.java:139) ~[opensearch-java-2.5.0.jar:?]
    at org.opensearch.dataprepper.plugins.sink.opensearch.dlq.FailedBulkOperationConverter.convertDocumentToGenericMap(FailedBulkOperationConverter.java:65) ~[opensearch-2.3.1.jar:?]
    at org.opensearch.dataprepper.plugins.sink.opensearch.dlq.FailedBulkOperationConverter.convertToDlqObject(FailedBulkOperationConverter.java:40) ~[opensearch-2.3.1.jar:?]
    at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195) ~[?:?]
    at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948) ~[?:?]
    at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484) ~[?:?]
    at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474) ~[?:?]
    at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913) ~[?:?]
    at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
    at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578) ~[?:?]
    at org.opensearch.dataprepper.plugins.sink.opensearch.OpenSearchSink.logFailureForBulkRequests(OpenSearchSink.java:328) ~[opensearch-2.3.1.jar:?]
    at org.opensearch.dataprepper.plugins.sink.opensearch.BulkRetryStrategy.handleFailures(BulkRetryStrategy.java:334) ~[opensearch-2.3.1.jar:?]
    at org.opensearch.dataprepper.plugins.sink.opensearch.BulkRetryStrategy.handleFailures(BulkRetryStrategy.java:252) ~[opensearch-2.3.1.jar:?]
    at org.opensearch.dataprepper.plugins.sink.opensearch.BulkRetryStrategy.handleRetriesAndFailures(BulkRetryStrategy.java:245) ~[opensearch-2.3.1.jar:?]
    at org.opensearch.dataprepper.plugins.sink.opensearch.BulkRetryStrategy.handleRetry(BulkRetryStrategy.java:269) ~[opensearch-2.3.1.jar:?]
    at org.opensearch.dataprepper.plugins.sink.opensearch.BulkRetryStrategy.execute(BulkRetryStrategy.java:191) ~[opensearch-2.3.1.jar:?]
    at org.opensearch.dataprepper.plugins.sink.opensearch.OpenSearchSink.lambda$flushBatch$6(OpenSearchSink.java:314) ~[opensearch-2.3.1.jar:?]
    at io.micrometer.core.instrument.composite.CompositeTimer.record(CompositeTimer.java:141) ~[micrometer-core-1.10.5.jar:1.10.5]
    at org.opensearch.dataprepper.plugins.sink.opensearch.OpenSearchSink.flushBatch(OpenSearchSink.java:311) ~[opensearch-2.3.1.jar:?]
    at org.opensearch.dataprepper.plugins.sink.opensearch.OpenSearchSink.doOutput(OpenSearchSink.java:283) ~[opensearch-2.3.1.jar:?]
    at org.opensearch.dataprepper.model.sink.AbstractSink.lambda$output$0(AbstractSink.java:64) ~[data-prepper-api-2.3.1.jar:?]
    at io.micrometer.core.instrument.composite.CompositeTimer.record(CompositeTimer.java:141) ~[micrometer-core-1.10.5.jar:1.10.5]
    at org.opensearch.dataprepper.model.sink.AbstractSink.output(AbstractSink.java:64) ~[data-prepper-api-2.3.1.jar:?]
    at org.opensearch.dataprepper.pipeline.Pipeline.lambda$publishToSinks$5(Pipeline.java:336) ~[data-prepper-core-2.3.1.jar:?]
```

There may be other areas where the assumption that action is `index` is made. The code should be audited to ensure all document accesses account for both supported actions.

**Expected behavior**
DLQ serialization works regardless of action

--- a/data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/BulkOperationWrapper.java
+++ b/data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/BulkOperationWrapper.java
@@ -7,9 +7,34 @@ package org.opensearch.dataprepper.plugins.sink.opensearch;
 
 import org.opensearch.client.opensearch.core.bulk.BulkOperation;
 import org.opensearch.dataprepper.model.event.EventHandle;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
 import static com.google.common.base.Preconditions.checkNotNull;
 
 public class BulkOperationWrapper {
+    private static final Predicate<BulkOperation> IS_INDEX_OPERATION = BulkOperation::isIndex;
+    private static final Predicate<BulkOperation> IS_CREATE_OPERATION = BulkOperation::isCreate;
+
+    private static final Map<Predicate<BulkOperation>, Function<BulkOperation, Object>> BULK_OPERATION_TO_DOCUMENT_CONVERTERS = Map.of(
+            IS_INDEX_OPERATION, operation -> operation.index().document(),
+            IS_CREATE_OPERATION, operation -> operation.create().document()
+    );
+
+    private static final Map<Predicate<BulkOperation>, Function<BulkOperation, String>> BULK_OPERATION_TO_INDEX_NAME_CONVERTERS = Map.of(
+            IS_INDEX_OPERATION, operation -> operation.index().index(),
+            IS_CREATE_OPERATION, operation -> operation.create().index()
+    );
+
+    private static final Map<Predicate<BulkOperation>, Function<BulkOperation, String>> BULK_OPERATION_TO_ID_CONVERTERS = Map.of(
+            IS_INDEX_OPERATION, operation -> operation.index().id(),
+            IS_CREATE_OPERATION, operation -> operation.create().id()
+    );
+
     private final EventHandle eventHandle;
     private final BulkOperation bulkOperation;
 
@@ -36,5 +61,30 @@ public class BulkOperationWrapper {
         if (eventHandle != null) {
             eventHandle.release(result);
         }
+    }
+
+    public Object getDocument() {
+        return getValueFromConverter(BULK_OPERATION_TO_DOCUMENT_CONVERTERS);
+    }
+
+    public String getIndex() {
+        return getValueFromConverter(BULK_OPERATION_TO_INDEX_NAME_CONVERTERS);
+    }
+
+    public String getId() {
+        return getValueFromConverter(BULK_OPERATION_TO_ID_CONVERTERS);
+    }
+
+    private <T> T getValueFromConverter(final Map<Predicate<BulkOperation>, Function<BulkOperation, T>> converters) {
+        final List<T> values = converters.entrySet().stream()
+                .filter(entry -> entry.getKey().test(bulkOperation))
+                .map(entry -> entry.getValue().apply(bulkOperation))
+                .collect(Collectors.toList());
+
+        if (values.size() != 1) {
+            throw new UnsupportedOperationException("Only index or create operations are supported currently." + bulkOperation);
+        }
+
+        return values.get(0);
     }
 }

--- a/data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/bulk/BulkOperationWriter.java
+++ b/data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/bulk/BulkOperationWriter.java
@@ -34,9 +34,9 @@ package org.opensearch.dataprepper.plugins.sink.opensearch.bulk;
 
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
-import org.opensearch.client.opensearch.core.bulk.BulkOperation;
 import org.opensearch.common.unit.ByteSizeValue;
 import org.opensearch.dataprepper.model.failures.DlqObject;
+import org.opensearch.dataprepper.plugins.sink.opensearch.BulkOperationWrapper;
 import org.opensearch.dataprepper.plugins.sink.opensearch.dlq.FailedDlqData;
 
 /**
@@ -47,10 +47,10 @@ public class BulkOperationWriter {
 
     private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
 
-    public static String bulkOperationToString(BulkOperation bulkOperation) {
-        String index = bulkOperation.index().index();
+    public static String bulkOperationToString(BulkOperationWrapper bulkOperation) {
+        String index = bulkOperation.getIndex();
         String source = extractDocumentSource(bulkOperation);
-        String id = bulkOperation.index().id();
+        String id = bulkOperation.getId();
 
         String sSource = "_na_";
         try {
@@ -86,8 +86,8 @@ public class BulkOperationWriter {
         }
     }
 
-    private static String extractDocumentSource(BulkOperation bulkOperation) {
-        final SerializedJson document = (SerializedJson) bulkOperation.index().document();
+    private static String extractDocumentSource(BulkOperationWrapper bulkOperation) {
+        final SerializedJson document = (SerializedJson) bulkOperation.getDocument();
 
         return new String(document.getSerializedJson());
     }

--- a/data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/bulk/JavaClientAccumulatingCompressedBulkRequest.java
+++ b/data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/bulk/JavaClientAccumulatingCompressedBulkRequest.java
@@ -126,15 +126,7 @@ public class JavaClientAccumulatingCompressedBulkRequest implements Accumulating
     }
 
     private Object mapBulkOperationToDocument(final BulkOperationWrapper bulkOperation) {
-        Object anyDocument;
-
-        if (bulkOperation.getBulkOperation().isIndex()) {
-            anyDocument = bulkOperation.getBulkOperation().index().document();
-        } else if (bulkOperation.getBulkOperation().isCreate()) {
-            anyDocument = bulkOperation.getBulkOperation().create().document();
-        } else {
-            throw new UnsupportedOperationException("Only index or create operations are supported currently. " + bulkOperation);
-        }
+        final Object anyDocument = bulkOperation.getDocument();
 
         if (anyDocument == null) {
             return new SerializedJsonImpl(null);

--- a/data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/bulk/JavaClientAccumulatingUncompressedBulkRequest.java
+++ b/data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/bulk/JavaClientAccumulatingUncompressedBulkRequest.java
@@ -69,16 +69,7 @@ public class JavaClientAccumulatingUncompressedBulkRequest implements Accumulati
     }
 
     private long estimateBulkOperationSize(BulkOperationWrapper bulkOperation) {
-
-        Object anyDocument;
-
-        if (bulkOperation.getBulkOperation().isIndex()) {
-            anyDocument = bulkOperation.getBulkOperation().index().document();
-        } else if (bulkOperation.getBulkOperation().isCreate()) {
-            anyDocument = bulkOperation.getBulkOperation().create().document();
-        } else {
-            throw new UnsupportedOperationException("Only index or create operations are supported currently. " + bulkOperation);
-        }
+        final Object anyDocument = bulkOperation.getDocument();
 
         if (anyDocument == null)
             return OPERATION_OVERHEAD;

--- a/data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/dlq/FailedBulkOperationConverter.java
+++ b/data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/dlq/FailedBulkOperationConverter.java
@@ -2,7 +2,6 @@ package org.opensearch.dataprepper.plugins.sink.opensearch.dlq;
 
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.google.common.collect.ImmutableMap;
-import org.opensearch.client.opensearch.core.bulk.BulkOperation;
 import org.opensearch.client.opensearch.core.bulk.BulkResponseItem;
 import org.opensearch.dataprepper.model.failures.DlqObject;
 import org.opensearch.dataprepper.plugins.sink.opensearch.bulk.SerializedJson;
@@ -34,14 +33,13 @@ public class FailedBulkOperationConverter {
     public DlqObject convertToDlqObject(final FailedBulkOperation failedBulkOperation) {
 
         final BulkOperationWrapper bulkOperationWithHandle = failedBulkOperation.getBulkOperation();
-        final BulkOperation bulkOperation = bulkOperationWithHandle.getBulkOperation();
         final BulkResponseItem bulkResponseItem = failedBulkOperation.getBulkResponseItem();
 
-        final Object document = convertDocumentToGenericMap(bulkOperation);
+        final Object document = convertDocumentToGenericMap(bulkOperationWithHandle);
 
         final FailedDlqData.Builder failedDlqDataBuilder = FailedDlqData.builder()
-            .withIndex(bulkOperation.index().index())
-            .withIndexId(bulkOperation.index().id())
+            .withIndex(bulkOperationWithHandle.getIndex())
+            .withIndexId(bulkOperationWithHandle.getId())
             .withDocument(document);
 
         if (bulkResponseItem != null) {
@@ -61,8 +59,8 @@ public class FailedBulkOperationConverter {
             .build();
     }
 
-    private Object convertDocumentToGenericMap(final BulkOperation bulkOperation) {
-        final SerializedJson document = (SerializedJson) bulkOperation.index().document();
+    private Object convertDocumentToGenericMap(final BulkOperationWrapper bulkOperation) {
+        final SerializedJson document = (SerializedJson) bulkOperation.getDocument();
         final byte[] documentBytes = document.getSerializedJson();
         final String jsonString = new String(documentBytes, StandardCharsets.UTF_8);
 

--- a/data-prepper-plugins/opensearch/src/test/java/org/opensearch/dataprepper/plugins/sink/opensearch/BulkOperationWrapperTests.java
+++ b/data-prepper-plugins/opensearch/src/test/java/org/opensearch/dataprepper/plugins/sink/opensearch/BulkOperationWrapperTests.java
@@ -5,18 +5,34 @@
 
 package org.opensearch.dataprepper.plugins.sink.opensearch;
 
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
 import org.opensearch.client.opensearch.core.bulk.BulkOperation;
+import org.opensearch.client.opensearch.core.bulk.CreateOperation;
+import org.opensearch.client.opensearch.core.bulk.DeleteOperation;
+import org.opensearch.client.opensearch.core.bulk.IndexOperation;
 import org.opensearch.dataprepper.model.event.EventHandle;
 import org.junit.jupiter.api.Test;
+
+import java.util.Objects;
+import java.util.UUID;
+import java.util.stream.Stream;
+
 import static org.hamcrest.CoreMatchers.equalTo;
 import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.mockito.Mockito.mock;
 
 public class BulkOperationWrapperTests {
+    private static final String ID = UUID.randomUUID().toString();
+    private static final String INDEX = UUID.randomUUID().toString();
+    private static final String DOCUMENT = UUID.randomUUID().toString();
+
     private BulkOperation bulkOperation;
 
-    BulkOperationWrapper createObjectUnderTest(EventHandle eventHandle) {
-        bulkOperation = mock(BulkOperation.class);
+    BulkOperationWrapper createObjectUnderTest(final EventHandle eventHandle, BulkOperation aBulkOperation) {
+        bulkOperation = Objects.isNull(aBulkOperation) ? mock(BulkOperation.class) : aBulkOperation;
         if (eventHandle == null) {
             return new BulkOperationWrapper(bulkOperation);
         }
@@ -25,7 +41,7 @@ public class BulkOperationWrapperTests {
 
     @Test
     public void testConstructorWithOneArgument() {
-        BulkOperationWrapper bulkOperationWithHandle = createObjectUnderTest(null);
+        BulkOperationWrapper bulkOperationWithHandle = createObjectUnderTest(null, null);
         assertThat(bulkOperationWithHandle.getBulkOperation(), equalTo(bulkOperation));
         assertThat(bulkOperationWithHandle.getEventHandle(), equalTo(null));
     }
@@ -33,8 +49,84 @@ public class BulkOperationWrapperTests {
     @Test
     public void testConstructorWithTwoArguments() {
         EventHandle eventHandle = mock(EventHandle.class);
-        BulkOperationWrapper bulkOperationWithHandle = createObjectUnderTest(eventHandle);
+        BulkOperationWrapper bulkOperationWithHandle = createObjectUnderTest(eventHandle, null);
         assertThat(bulkOperationWithHandle.getBulkOperation(), equalTo(bulkOperation));
         assertThat(bulkOperationWithHandle.getEventHandle(), equalTo(eventHandle));
+    }
+
+    @ParameterizedTest
+    @MethodSource("bulkOperationProvider")
+    public void testGetId(final BulkOperation bulkOperation) {
+        final BulkOperationWrapper bulkOperationWrapper = createObjectUnderTest(null, bulkOperation);
+        assertThat(bulkOperationWrapper.getId(), equalTo(ID));
+    }
+
+    @Test
+    public void testGetIdUnsupportedAction() {
+        final BulkOperationWrapper bulkOperationWrapper = createObjectUnderTest(null, getDeleteBulkOperation());
+        assertThrows(UnsupportedOperationException.class, bulkOperationWrapper::getId);
+    }
+
+    @ParameterizedTest
+    @MethodSource("bulkOperationProvider")
+    public void testGetIndex(final BulkOperation bulkOperation) {
+        final BulkOperationWrapper bulkOperationWrapper = createObjectUnderTest(null, bulkOperation);
+        assertThat(bulkOperationWrapper.getIndex(), equalTo(INDEX));
+    }
+
+    @Test
+    public void testGetIndexUnsupportedAction() {
+        final BulkOperationWrapper bulkOperationWrapper = createObjectUnderTest(null, getDeleteBulkOperation());
+        assertThrows(UnsupportedOperationException.class, bulkOperationWrapper::getIndex);
+    }
+
+    @ParameterizedTest
+    @MethodSource("bulkOperationProvider")
+    public void testGetDocument(final BulkOperation bulkOperation) {
+        final BulkOperationWrapper bulkOperationWrapper = createObjectUnderTest(null, bulkOperation);
+        assertThat(bulkOperationWrapper.getDocument(), equalTo(DOCUMENT));
+    }
+
+    @Test
+    public void testGetDocumentUnsupportedAction() {
+        final BulkOperationWrapper bulkOperationWrapper = createObjectUnderTest(null, getDeleteBulkOperation());
+        assertThrows(UnsupportedOperationException.class, bulkOperationWrapper::getDocument);
+    }
+
+    private static Stream<Arguments> bulkOperationProvider() {
+        final IndexOperation indexOperation = new IndexOperation.Builder<>()
+                .id(ID)
+                .index(INDEX)
+                .document(DOCUMENT)
+                .build();
+        final BulkOperation indexBulkOperation = (BulkOperation) new BulkOperation.Builder()
+                .index(indexOperation)
+                .build();
+
+        final CreateOperation createOperation = new CreateOperation.Builder<>()
+                .id(ID)
+                .index(INDEX)
+                .document(DOCUMENT)
+                .build();
+        final BulkOperation createBulkOperation = (BulkOperation) new BulkOperation.Builder()
+                .create(createOperation)
+                .build();
+
+        return Stream.of(
+                Arguments.of(
+                        indexBulkOperation,
+                        createBulkOperation
+                )
+        );
+    }
+
+    private BulkOperation getDeleteBulkOperation() {
+        final DeleteOperation deleteOperation = new DeleteOperation.Builder()
+                .id(ID)
+                .index(INDEX)
+                .build();
+        return new BulkOperation.Builder()
+                .delete(deleteOperation)
+                .build();
     }
 }

--- a/data-prepper-plugins/opensearch/src/test/java/org/opensearch/dataprepper/plugins/sink/opensearch/dlq/FailedBulkOperationConverterTest.java
+++ b/data-prepper-plugins/opensearch/src/test/java/org/opensearch/dataprepper/plugins/sink/opensearch/dlq/FailedBulkOperationConverterTest.java
@@ -6,6 +6,7 @@ import org.junit.jupiter.api.Test;
 import org.opensearch.client.opensearch._types.ErrorCause;
 import org.opensearch.client.opensearch.core.bulk.BulkOperation;
 import org.opensearch.client.opensearch.core.bulk.BulkResponseItem;
+import org.opensearch.client.opensearch.core.bulk.CreateOperation;
 import org.opensearch.client.opensearch.core.bulk.IndexOperation;
 import org.opensearch.dataprepper.model.failures.DlqObject;
 import org.opensearch.dataprepper.plugins.sink.opensearch.bulk.SerializedJson;
@@ -52,11 +53,22 @@ public class FailedBulkOperationConverterTest {
         failureMessage = UUID.randomUUID().toString();
         failure = new Exception(failureMessage);
         bulkOperation = mock(BulkOperation.class);
+
         final IndexOperation indexOperation = mock(IndexOperation.class);
-        when(bulkOperation.index()).thenReturn(indexOperation);
         when(indexOperation.index()).thenReturn(testIndex);
         when(indexOperation.document()).thenReturn(document);
         when(indexOperation.id()).thenReturn(testId);
+
+        final CreateOperation createOperation = mock(CreateOperation.class);
+        when(createOperation.index()).thenReturn(testIndex);
+        when(createOperation.document()).thenReturn(document);
+        when(createOperation.id()).thenReturn(testId);
+
+        when(bulkOperation.isIndex()).thenReturn(true);
+        when(bulkOperation.index()).thenReturn(indexOperation);
+        when(bulkOperation.isCreate()).thenReturn(false);
+        when(bulkOperation.create()).thenReturn(createOperation);
+
         final ErrorCause errorCause = mock(ErrorCause.class);
         bulkResponseItem = mock(BulkResponseItem.class);
         when(bulkResponseItem.status()).thenReturn(RestStatus.INTERNAL_SERVER_ERROR.getStatus());
@@ -74,6 +86,22 @@ public class FailedBulkOperationConverterTest {
             .withBulkResponseItem(bulkResponseItem)
             .withFailure(failure)
             .build();
+
+        final DlqObject result = converter.convertToDlqObject(testData);
+
+        validateResponse(result, errorReason);
+    }
+
+    @Test
+    public void testConvertActionCreateToDlqObject() {
+        when(bulkOperation.isIndex()).thenReturn(false);
+        when(bulkOperation.isCreate()).thenReturn(true);
+
+        final FailedBulkOperation testData = FailedBulkOperation.builder()
+                .withBulkOperation(new BulkOperationWrapper(bulkOperation))
+                .withBulkResponseItem(bulkResponseItem)
+                .withFailure(failure)
+                .build();
 
         final DlqObject result = converter.convertToDlqObject(testData);
 

